tool/tctl/common/tctl.go

```diff
@@ -196,9 +196,10 @@ func TryRun(commands []CLICommand, args []string) error {
 
 	ctx := context.Background()
 
-	mfaPrompt := mfa.NewPrompt("")
-	mfa.WithPromptReasonAdminAction()(mfaPrompt)
-	clientConfig.PromptAdminRequestMFA = mfaPrompt.Run
+	promptCfg := mfa.DefaultPromptConfig("")
+	mfa.WithPromptReasonAdminAction()(promptCfg)
```

Comment:
Could we apply the options directly to the constructor, or is it too much hassle? This works, but it doesn't function like an option.

Reply:
The issue is that we don't get the proxy address until after the client is created. But I can instead add a `client.SetMFAPrompConstructor` to get around this awkward logic.

Reply:
Sorry I completely misunderstood your question, but I actually already did what you're asking. The PromptOpt logic is now being applied in place where the mfa prompt is being used:
```go
// RetryWithMFAUnaryInterceptor intercepts a GRPC client unary call to check if the
// error indicates that the client should retry with MFA verification.
func RetryWithMFAUnaryInterceptor(mfaCeremony func(ctx context.Context, opts ...mfa.PromptOpt) (*proto.MFAAuthenticateResponse, error)) grpc.UnaryClientInterceptor {
	return func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
		err := invoker(ctx, method, req, reply, cc, opts...)
		if !errors.Is(trail.FromGRPC(err), &mfa.ErrAdminActionMFARequired) {
			return err
		}

		mfaResp, ceremonyErr := mfaCeremony(ctx, mfa.WithPromptReasonAdminAction(method))
		if ceremonyErr != nil {
			return trace.NewAggregate(trail.FromGRPC(err), ceremonyErr)
		}

		return invoker(ctx, method, req, reply, cc, append(opts, mfa.WithCredentials(mfaResp))...)
	}
}
```